controlplane/api/v1beta1/rke2controlplane_types.go

```diff
@@ -72,11 +71,14 @@ type RKE2ControlPlaneSpec struct {
 
 	// InfrastructureRef is a required reference to a custom resource
 	// offered by an infrastructure provider.
+	// This field is deprecated. Use `.machineTemplate.infrastructureRef` instead.
```

Comment:
we can add `// +kubebuilder:deprecatedversion` tag
https://book.kubebuilder.io/reference/markers/crd